provider/sgx.py

```diff
@@ -0,0 +1,200 @@
+            if node_size != '0':
+                node_list.append(str(i))
+        monitor_expect_nodes = int(self._params["monitor_expect_nodes"])
+        if len(node_list) < monitor_expect_nodes:
+            self._test.cancel("host numa nodes %s isn't enough for "
+                              "testing." % node_list)
+
+
+class SGXChecker(object):
+    """
+    General operations for checking guest sgx enablement.
+    """
+
+    def __init__(self, test, params, vm):
+        """
+
+        :param test: Context of test.
+        :param params: params of running ENV.
+        :param vm:VM object.
+        """
+        self._test = test
+        self._params = params
+        self._vm = vm
+        self._monitor = vm.monitor
+
+    def verify_sgx_flags(self, qmp_command, flags):
+        """
+        Check if sgx cpu flags enabled in qmp cmd
+
+        :param qmp_command: query sgx qmp command output
+        :param flags: sgx flags need to be verified
+        """
+        for ele in flags:
+            if qmp_command[ele] is not True:
+                self._test.fail("%s is not enabled, qmp check failed." % ele)
+
+    def get_config_total_epc_size(self):
+        """
+        Get total epc size from configuration.
+
+        :return: total configured epc size
+        """
+        epc_list = self._params["vm_sgx_epc_devs"].split()
+        config_epc_size = 0
+        for ele in epc_list:
+            epc_params = self._params.object_params(ele)
+            ele_memdev = self._params.object_params(
+                epc_params["vm_sgx_epc_memdev"])
+            tmp_epc_size = int(float(normalize_data_size(
+                ele_memdev["size_mem"], "B", 1024)))
+            config_epc_size += tmp_epc_size
+        return config_epc_size
+
+    def get_config_epc_numa_info(self):
+        """
+        Get epc size on each nume node from configuration.
+
+        :return: epc size on each nume node configured.
+        """
+        guest_sgx_epc_list = self._params["vm_sgx_epc_devs"].split()
+        tmp_epc_dict = {}
+        for ele in guest_sgx_epc_list:
+            epc_params = self._params.object_params(ele)
+            ele_memdev = self._params.object_params(
+                epc_params["vm_sgx_epc_memdev"])
+            tmp_epc_size = int(float(normalize_data_size(
+                ele_memdev["size_mem"], "B", 1024)))
+            epc_numa_id = int(epc_params["vm_sgx_epc_node"])
+            tmp_epc_dict = {'size': tmp_epc_size, 'node': epc_numa_id}
+        return tmp_epc_dict
+
+    def verify_qmp_host_sgx_cap(self):
+        """
+        Verify query host sgx capabilities qmp cmd in sgx flags and epc size
+        """
+        sgx_flags = self._params['sgx_flags'].split()
+        host_sgx_info = self._monitor.query_sgx_capabilities()
+        self.verify_sgx_flags(host_sgx_info, sgx_flags)
+
+        host_cap = SGXHostCapability(self._test, self._params)
+        host_epc_size = host_cap.host_epc_size
```

Comment:
verify_qmp_host_sgx_cap(self, host_epc_size)

We need to validate host sgx caps first in case, 
then we get the host_cap,
we can use the cap in the following checks

Reply:
If you want to keep consistent with guest sgx cap check function, we may pass flags and size to the function, as arguments, e.g.

verify_qmp_host_sgx_cap(self, expected_sgx_flags, expected_epc_size)
verify_qmp_guest_sgx_cap(self, expected_sgx_flags, expected_epc_size)

As you encapsulate the related into a public function, we can get it in case and pass it to the verify* function

Reply:
I get it, thanks. host_epc_size will be an argument passed to verify_qmp_host_sgx_cap.
Since this 2 functions verify_qmp_host_sgx_cap() and verify_qmp_guest_sgx_cap() verified test points are not 100% same, I will only pass argument to host qmp function.